packages/nimble-components/src/wafer-map/tests/experimental-data-manager.spec.ts

```diff
@@ -0,0 +1,162 @@
+            width: Math.ceil(dataManagerModule.dieDimensions.width),
+            height: Math.ceil(dataManagerModule.dieDimensions.height)
+        };
+        expect(computedDimensions).toEqual({
+            width: 19,
+            height: 16
+        });
+    });
+
+    it('should have expected margin', () => {
+        expect(dataManagerModule.margin).toEqual(expectedMargin);
+    });
+
+    it('should have increasing horizontal range', () => {
+        expect(dataManagerModule.horizontalScale.range()).toEqual([0, 92]);
+    });
+
+    it('should have decreasing vertical range', () => {
+        // because the canvas has top-left origin location we need to flip the vertical scale
+        expect(dataManagerModule.verticalScale.range()).toEqual([92, 0]);
+    });
+
+    it('should not have labelsFontSize larger than the die height', () => {
+        expect(dataManagerModule.labelsFontSize).toBeLessThanOrEqual(
+            dataManagerModule.dieDimensions.height
+        );
+    });
+
+    it('should not have labelsFontSize larger than the die width', () => {
+        expect(dataManagerModule.labelsFontSize).toBeLessThanOrEqual(
+            dataManagerModule.dieDimensions.width
+        );
+    });
+
+    it('should have as many dies as provided', () => {
+        expect(dataManagerModule.diesRenderInfo.length).toEqual(
+            getWaferMapDies().length
+        );
+    });
+
+    it('should have label with suffix for each die', () => {
+        for (const dieInfo of dataManagerModule.diesRenderInfo) {
+            expect(dieInfo.text).toContain(dieLabelsSuffix);
+        }
+    });
+
+    // skipped until prerendering is refactored
+    xit('should have all dies with full opacity from the highlighted list', () => {
+        const highlightedTags = getHighlightedTags();
+        const dies = getWaferMapDies().filter(die => die.tags?.some(dieTag => highlightedTags.some(
+            highlightedTag => dieTag === highlightedTag
+        )));
+        const diesWithFullOpacity = dataManagerModule.diesRenderInfo.filter(x => x.fillStyle.endsWith(',1)'));
+        expect(dies.length).toEqual(diesWithFullOpacity.length);
+    });
+
+    // skipped until prerendering is refactored
+    xit('should not have any dies with partial opacity from the highlighted list', () => {
+        const highlightedTags = getHighlightedTags();
+        const dies = getWaferMapDies().filter(
+            die => !die.tags?.some(dieTag => highlightedTags.some(
+                highlightedTag => dieTag === highlightedTag
+            ))
+        );
+        const diesWithPartialOpacity = dataManagerModule.diesRenderInfo.filter(
+            x => !x.fillStyle.endsWith(',1)')
+        );
+        expect(dies.length).toEqual(diesWithPartialOpacity.length);
+    });
+
+    // skipped until prerendering is refactored
+    xit('should have all dies inside the canvas with margins', () => {
+        for (const dieRenderInfo of dataManagerModule.diesRenderInfo) {
```

Comment:
This test should [not have conditional asserts](https://github.com/ni/javascript-styleguide/issues/125#issuecomment-1631721812) from looping over a structure not defined in the test. Instead precalculate the expected array and assert the array at once. If not cleaning up now include in the "skipped" comment that it needs to be cleaned up.